File: AoC2021.FSharp/Day0.fs
namespace AoC2021.FSharp

open AoC2021.Core
open Xunit

module Day0 =
    let Solve =
        Seq.fold (fun acc item -> acc * int64 item) 1L

    [<Fact>]
    let ``Solves Example`` () =
        let exampleInput = @"foo"

        let actual = Solve exampleInput
        Assert.Equal(1256742L, actual)

type Day0(input: string) =
    interface ISolver with
        member this.SolvePartOne() = Day0.Solve input
        member this.SolvePartTwo() = failwith "NotImplemented"

File: AoC2021.CSharp/Day0.cs
using AoC2021.Core;

using Xunit;

namespace AoC2021.CSharp;

public class Day0 : Solver {
    public Day0(string? input) : base(input) { }

    public override string SolvePartOne() {
        return "Day 0 C#: " + Input;
    }

    [Fact]
    public void SolvesExample() {
        const string? exampleInput = @"foo";

        var solver = new Day0(exampleInput);
        var actual = solver.SolvePartOne();

        Assert.Equal("Day 0 C#: foo", actual);
    }
}